packages/cli/src/CredentialsHelper.ts

```diff
@@ -786,15 +786,9 @@ export class CredentialsHelper extends ICredentialsHelper {
 
 		const credential = await this.sharedCredentialsRepository.findOne({
 			where: {
-				role: {
-					scope: 'credential',
-					name: 'owner',
-				},
+				role: 'owner',
 				user: {
-					globalRole: {
-						scope: 'global',
-						name: 'owner',
-					},
+					role: 'owner',
```

Comment:
I like this idea, [so I tried doing this](https://github.com/n8n-io/n8n/commit/70dc338901a75eb5b2044005624e443f8eb19633), but the code gets a lot more verbose IMO.

In the current form, the IDE can already tell us what kind of role this is 
![image](https://github.com/n8n-io/n8n/assets/196144/adff21d8-6bc6-4efb-9f0f-2fcf69219605)

Maybe @krynble or @ivov have opinions on this?

Reply:
My concern is also that with the roles being plain (but typed) strings that something may slip through as the wrong type of role. It's probably unlikely due to typing but I'd rather be safe when it comes to permissions.

Reply:
I like the idea of the prefixes because it clearly states what we are seeing.

Even though the IDE can auto complete stuff just fine, when viewing a PR in Github it's not possible, and what I'd like to do is avoid confusing, in the future, the role types / names.

Right now we have `editor` for workflow but `user` for credential which could open up for more differences in the future, so knowing clearly that you picked the right value could be helpful unless wen can guarantee that it won't happen due to type safety.

Reply:
I still like the idea, but the code still looks verbose to me.
However, I've pushed a change updating the role names everywhere, and all tests seem to still pass.
I've also tested migrating up and down on all 3 DBs, and this change looks to be fairly reversible (in case we decide to revert this due to a bug).